Add cluster management commands [CLC-163] (#231)
hazelcast/hazelcast-commandline-client: docs/modules/ROOT/pages/clc-viridian.adoc

== Commands

* <<clc-viridian-login, clc viridian login>>
* <<clc-viridian-create-cluster, clc viridian create-cluster>>
* <<clc-viridian-get-cluster, clc viridian get-cluster>>
* <<clc-viridian-list-clusters, clc viridian list-clusters>>

> * <<clc-viridian-stop-cluster, clc viridian get-cluster>>

Reviewer: ```suggestion
* <<clc-viridian-stop-cluster, clc viridian stop-cluster>>
```

> * <<clc-viridian-resume-cluster, clc viridian get-cluster>>

Reviewer: ```suggestion
* <<clc-viridian-resume-cluster, clc viridian resume-cluster>>
```

> * <<clc-viridian-delete-cluster, clc viridian get-cluster>>

Reviewer: ```suggestion
* <<clc-viridian-delete-cluster, clc viridian delete-cluster>>
```

|===

== clc viridian create-cluster

> Creates a Viridian cluster for the logged in API key.

Reviewer: ```suggestion
Creates a Viridian cluster.
```

Make sure you login before running this command.

Usage:

[source,bash]
----
clc viridian create-cluster [flags]
----

Parameters:

[cols="1m,1a,2a,1a"]
|===
|Parameter|Required|Description|Default

|`--api-key`
|Optional
|Sets the API key. Overrides the `CLC_VIRIDIAN_API_KEY` environment variable. If not given, one of the existing api keys will be used.
|

|`--name`
|Optional

> |Sets the name of the created cluster. If not given, random name will be generated.

Reviewer: ```suggestion
|Sets the name of the created cluster. If not given, an auto-generated name will be used.
```

|

|`--plan`
|Optional
|Sets the plan for the created cluster.
|serverless

|`--development`
|Optional
|Creates a development cluster.
|false
|===

== clc viridian get-cluster

> Gets the information about the given Viridian cluster for the logged in API key.

Reviewer: ```suggestion
Gets the information about the given Viridian cluster.
```

clc viridian list-clusters [flags]
----

Parameters:

[cols="1m,1a,2a,1a"]
|===
|Parameter|Required|Description|Default

|`--api-key`
|Optional
|Sets the API key. Overrides the `CLC_VIRIDIAN_API_KEY` environment variable. If not given, one of the existing api keys will be used.
|

|===

== clc viridian stop-cluster

> Stops the given Viridian cluster for the logged in API key.

Reviewer: ```suggestion
Stops the given Viridian cluster.
```

Make sure you login before running this command.

Usage:

[source,bash]
----
clc viridian stop-cluster [cluster-ID/name] [flags]
----

Parameters:

[cols="1m,1a,2a,1a"]
|===
|Parameter|Required|Description|Default

|`--api-key`
|Optional
|Sets the API key. Overrides the `CLC_VIRIDIAN_API_KEY` environment variable. If not given, one of the existing api keys will be used.
|

|===

== clc viridian resume-cluster

> Resumes the given Viridian cluster for the logged in API key.

Reviewer: ```suggestion
Resumes the given Viridian cluster.
```

Make sure you login before running this command.

Usage:

[source,bash]
----
clc viridian resume-cluster [cluster-ID/name] [flags]
----

Parameters:

[cols="1m,1a,2a,1a"]
|===
|Parameter|Required|Description|Default

|`--api-key`
|Optional
|Sets the API key. Overrides the `CLC_VIRIDIAN_API_KEY` environment variable. If not given, one of the existing api keys will be used.
|

|===

== clc viridian delete-cluster

> Deletes the given Viridian cluster for the logged in API key.

Reviewer: ```suggestion
Deletes the given Viridian cluster.
```

Make sure you login before running this command.

Usage:

[source,bash]
----
clc viridian delete-cluster [cluster-ID/name] [flags]
----

Parameters:

[cols="1m,1a,2a,1a"]
|===
|Parameter|Required|Description|Default

|`--api-key`
|Optional
|Sets the API key. Overrides the `CLC_VIRIDIAN_API_KEY` environment variable. If not given, one of the existing api keys will be used.
|

|`--yes`
|Optional
|Skips confirming the delete operation.
|

|===

== clc viridian download-logs

> Downloads the logs of the given Viridian cluster for the logged in API key.

Reviewer: ```suggestion
Downloads the logs of the given Viridian cluster.
```

> Downloads the logs of the given Viridian cluster for the logged in API key.

Reviewer: ```suggestion
Downloads the logs of the given Viridian cluster.
```
